plugins/toolbox-search/src/block_searcher.ts

```diff
@@ -27,29 +27,44 @@ export class BlockSearcher {
     const blockCreationWorkspace = new Blockly.Workspace();
     blockTypes.forEach((blockType) => {
       const block = blockCreationWorkspace.newBlock(blockType);
-      this.indexBlockText(blockType.replaceAll('_', ' '), blockType);
-      block.inputList.forEach((input) => {
-        input.fieldRow.forEach((field) => {
-          this.indexDropdownOption(field, blockType);
-          this.indexBlockText(field.getText(), blockType);
+      const blockState = Blockly.serialization.blocks.save(block);
+      if (blockState != null) {
+        this.indexBlockText(blockType.replaceAll('_', ' '), blockState);
+        block.inputList.forEach((input) => {
+          input.fieldRow.forEach((field) => {
+            this.indexDropdownOption(field, blockState);
+            this.indexBlockText(field.getText(), blockState);
+          });
         });
-      });
+      }
     });
   }
 
   /**
    * Check if the field is a dropdown, and index every text in the option
    *
    * @param field We need to check the type of field
-   * @param blockType The block type to associate the trigrams with.
+   * @param blockState The block state to associate the trigrams with.
    */
-  private indexDropdownOption(field: Blockly.Field, blockType: string) {
+  private indexDropdownOption(
+    field: Blockly.Field,
+    blockState: Blockly.serialization.blocks.State,
+  ) {
     if (field instanceof Blockly.FieldDropdown) {
       field.getOptions(true).forEach((option) => {
+        const state = {...blockState};
+        state.fields = {...blockState.fields};
         if (typeof option[0] === 'string') {
-          this.indexBlockText(option[0], blockType);
+          if (state.fields == undefined) {
+            state.fields = {};
+          }
+          if (field.name) {
+            state.fields[field.name] = option[1];
+          }
+          this.indexBlockText(option[0], state);
+          this.indexBlockText(option[1], state);
```

Comment:
I don't think we should be indexing the language-neutral text.

Reply:
Well, I added this on some blocks the language-neutral text was more informative than the block text. I don't remember now a clear example, but I take a look. BTW, sorry for replaying so late, but lately I am little busy with an internal project. I would try to find some time to fix your comments.